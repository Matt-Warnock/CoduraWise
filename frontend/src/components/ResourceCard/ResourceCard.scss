.resource-card {
  &__container {
    font-family: monospace, sans-serif;
    font-size: 0.9em;
    width: auto;
    max-width: 60em;
    height: 4em;
    margin: 10px 20px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    background-color: #1d3557;
    padding: 2%;
    border-radius: 3px;
    -webkit-box-shadow: 5px 2px 12px -7px rgba(112, 108, 112, 1);
    -moz-box-shadow: 5px 2px 12px -7px rgba(112, 108, 112, 1);
    box-shadow: 5px 2px 12px -7px rgba(112, 108, 112, 1);
  }

  &__link {
    text-decoration: underline;
    color: white;
    text-decoration-color: transparent;

    &:hover {
      text-decoration: underline;
      text-decoration-color: white;
      transition: 0.3s;
    }
  }
}
