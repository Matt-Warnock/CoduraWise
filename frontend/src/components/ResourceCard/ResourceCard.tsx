import React from "react";
import StarRating from "../StarRating/StarRating";
import "./ResourceCard.scss";

interface ResourceCardProps {
  title: string;
  link: string;
  rating: number;
}

const ResourceCard = ({ title, link, rating }: ResourceCardProps) => {
  return (
    <article className="resource-card__container">
      <h3 className="resource-card__title">
        <a
          className="resource-card__link"
          href={link}
          target="_blank"
          rel="noreferrer"
        >
          {title}
        </a>
      </h3>
      <StarRating rating={rating} />
    </article>
  );
};

export default ResourceCard;
