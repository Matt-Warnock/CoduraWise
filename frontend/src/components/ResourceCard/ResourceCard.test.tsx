import TestRenderer from "react-test-renderer";
import ResourceCard from "./ResourceCard";
import React from "react";

test("snapshot test: the html matches", () => {
  const card = TestRenderer.create(
    <ResourceCard link="url" title="title" rating={1} />,
  );

  expect(card.toJSON()).toMatchSnapshot();
});
