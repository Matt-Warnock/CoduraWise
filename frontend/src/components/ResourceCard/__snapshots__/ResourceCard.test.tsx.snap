// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: the html matches 1`] = `
<article
  className="resource-card__container"
>
  <h3
    className="resource-card__title"
  >
    <a
      className="resource-card__link"
      href="url"
      rel="noreferrer"
      target="_blank"
    >
      title
    </a>
  </h3>
  <div
    aria-label="Rating of this product is 1 out of 5."
    className="star-container"
    style={
      Object {
        "--rating": 1,
      }
    }
  />
</article>
`;
